### PR TITLE
Xenoarch analysis machines NanoUI + hotkeys.

### PR DESCRIPTION
**Pull request by PJB3005**
Also a new sprite for the Hyperspectral Imager.

Crtl-click: start scanning.
Alt-click: eject beaker.
Also an alt click for the bunsen burner to toggle it.

Oh yeah did I mention the massive amounts of cleanup?

:cl:
 * imageadd: The Hyperspectral Imager now has its own unique sprite and doesn't look like a literal photocopier.
 * rscadd: Xenoarch analysis machines now use a fancy NanoUI.
 * rscadd: Xenoarch analysis machines have a ctrl+click shortcut for starting scanning and an alt+click shortcut for ejecting the inserted container.
 * rscadd: Bunsen burners are now also toggle with alt+click.

### CODE REVIEW

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 1]
```diff
@@ -100,115 +100,120 @@
...
-		dat += "[held_container ? "<A href='?src=\ref[src];begin=1'>Begin scanning</a>" : ""]"
-
-	dat += {"<hr>
-		<A href='?src=\ref[src];refresh=1'>Refresh</a><BR>
-		<A href='?src=\ref[src];close=1'>Close</a><BR>"}
-	user << browse(dat, "window=anomaly;size=450x500")
-	onclose(user, "anomaly")
-
-obj/machinery/anomaly/attackby(obj/item/weapon/W as obj, mob/living/user as mob)
+/obj/machinery/anomaly/attackby(obj/item/weapon/W as obj, mob/living/user as mob)
```
Comment (DamianX) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm]: `as`

File: code/modules/research/xenoarchaeology/machinery/analysis_base.dm [hunk 2]
```diff
@@ -24,61 +26,59 @@
...
-							//temp greater than 600 also requires a safety prompt to initiate scanning
-	var/max_temp = 450
 
-/obj/machinery/anomaly/New()
-	..()
+	//measured in kelvin, if this exceeds 1200, the machine is damaged and requires repairs
+	//if this exceeds 600 and safety is enabled it will shutdown
+	//temp greater than 600 also requires a safety prompt to initiate scanning
+	var/temperature = T0C
+	var/max_temp = 450
```
Comment (DamianX) [inline code/modules/research/xenoarchaeology/machinery/analysis_base.dm]: This seems to be unused

**Comment by Knightkiller11** [comment]
Didn't you make a PR like this back in April? Never really gave a reason why you closed it at the time.
That aside, this is looking good. Might actually dive back into Xenoarch for a bit now.

**Comment by PJB3005** [comment]
Oh I did PR it? Huh.

I don't recall why I closed it either probably drama.

**Comment by SkowronX** [comment]
>copy machine sprite

lol

**Review by Shadowmech88 — APPROVED** [review]

**Comment by Shadowmech88** [comment]
Shouldn't the UI changes and Alt/Ctrl+click changes be in the changelog as well?

**Pull request merged**